storage/remote/client.go

```diff
@@ -97,5 +120,5 @@ func (c *Client) Store(samples model.Samples) error {
 // will simply be removed in the restructuring and the whole "generic" naming
 // will be gone for good.
 func (c Client) Name() string {
-	return "generic"
+	return fmt.Sprintf("generic:%s", c.url)
```

Comment:
Differentiating only by endpoint in the metrics is not strictly enough as long as we don't prevent configuring the same endpoint twice. Currently the user can produce invalid metrics like this by doing that:

```
# HELP prometheus_remote_storage_queue_capacity The capacity of the queue of samples to be sent to the remote storage.
# TYPE prometheus_remote_storage_queue_capacity gauge
prometheus_remote_storage_queue_capacity{type="generic:http://localhost:1000/"} 102400
prometheus_remote_storage_queue_capacity{type="generic:http://localhost:1000/"} 102400
# HELP prometheus_remote_storage_queue_length The number of processed samples queued to be sent to the remote storage.
# TYPE prometheus_remote_storage_queue_length gauge
prometheus_remote_storage_queue_length{type="generic:http://localhost:1000/"} 0
prometheus_remote_storage_queue_length{type="generic:http://localhost:1000/"} 102400
```

Maybe also add an index to the name?